src/main/kotlin/org/jetbrains/research/libsl/visitors/FunctionVisitor.kt

```diff
@@ -94,16 +131,18 @@ class FunctionVisitor(
             expressionVisitor.visitExpression(expr)
         }.orEmpty().toMutableList()
 
+        context.storeAnnotation(Annotation(name, args))
+
         if (name == "target") {
-            return TargetAnnotation()
+            return AnnotationReference("target", context)
```

Comment:
Could you refactor out `@Target` annotation to some singleton variable? This could help in future refactoring